### PR TITLE
fix(achievementInfo): prevent adding types to subsets and test kits

### PR DESCRIPTION
**Pull request by wescopeland**
Related to #2014.

This PR prevents the user from adding types to an achievement if it is a subset or a test kit.

To test:
* You should be able to add types to `/achievement/9`.
* You should get an error if you try to add a type to `/achievement/241312`.
* You should always be able to remove a type, regardless of the game title.

### CODE REVIEW

File: public/achievementInfo.php [hunk 1]
```diff
@@ -121,8 +121,21 @@ function generateAchievementMetaDescription(
...
+    ?>
+
     <script>
     function updateAchievementDetails() {
+        const canHaveTypes = <?= $canHaveTypes ? 'true' : 'false' ?>;
+
+        const typeInputValue = $('#typeinput').val();
+
+        if (!canHaveTypes && typeInputValue) {
+            showStatusFailure('Error: Types are not supported on subsets or test kits.');
```
Comment (Jamiras) [inline public/achievementInfo.php:135]: Recommend hiding the dropdown if the type is already None

File: public/achievementInfo.php [hunk 2]
```diff
@@ -121,8 +121,21 @@ function generateAchievementMetaDescription(
 RenderContentStart($pageTitle);
 ?>
 <?php if ($permissions >= Permissions::Developer || ($permissions >= Permissions::JuniorDeveloper && $isAuthor)): ?>
+    <?php
+        $canHaveTypes = mb_strpos($gameTitle, "[Subset") === false && mb_strpos($gameTitle, "~Test Kit~") === false;
```
Comment (Jamiras) [inline public/achievementInfo.php:125]: Suggest extending this to Events

**Review by Jamiras — COMMENTED** [review]
Recommend hiding the dropd

File: public/achievementInfo.php (hunk 2 above)
Comment (wescopeland) [reply public/achievementInfo.php]: Updated to check for games that have a console ID of 101.

File: public/achievementInfo.php (hunk 1 above)
Comment (wescopeland) [reply public/achievementInfo.php]: Removed this new JS and set the dropdown to never appear if `$canHaveTypes` is falsy. Related SQL query we can execute after shipping: https://discord.com/channels/476211979464343552/1174126386349342901

**Review by Jamiras — APPROVED** [review]

**Pull request merged**